### ISSUE TITLE
#[doc(hidden)] pub Type with non hidden aliases not flagegd when breaking semver

### ISSUE THREAD

**Issue by jalil-salame** [obi1kenobi/cargo-semver-checks#895]
### Is this about an existing lint, or proposing a new one?

I guess this is from an existing lint? But I don't know which one...

### Known issues that might be causing this

- [ ] Is the item that should have been flagged originating from another crate, or from a re-export of such an item? ([#638](https://github.com/obi1kenobi/cargo-semver-checks/issues/638))
- [ ] Is the issue related to a change in the type of a field, function parameter, generic, or trait bound? ([#149](https://github.com/obi1kenobi/cargo-semver-checks/issues/149))
- [ ] If you're proposing a new lint, is it already part of our lint wishlist? ([#5](https://github.com/obi1kenobi/cargo-semver-checks/issues/5))

### Steps to reproduce the bug with the above code

Original code:

```rust
//! src/lib.rs
#[doc(hidden)]
pub struct IdType<T>(pub T);

pub type Id = IdType<String>;
```

Updated code (patch or minor release)):

```rust
//! src/lib.rs
#[doc(hidden)]
pub struct IdType<T>(T); // change visibility

pub type Id = IdType<String>; // Observable through the type alias
```

### Actual Behaviour

`cargo semver-checks` (v0.34.0) output:

```console
$ cargo semver-checks --baseline-rev '...'
     Cloning dddfce3f349b
     Parsing repro v0.1.1 (current)
      Parsed [   0.217s] (current)
     Parsing repro v0.1.0 (baseline)
      Parsed [   0.234s] (baseline)
    Checking repro v0.1.0 -> v0.1.1 (minor change)
     Checked [   0.002s] 72 checks: 72 pass, 7 skip
     Summary no semver update required
    Finished [   0.456s] repro
```

### Expected Behaviour

`cargo semver-checks` should flag the visibility change in `IdType.0` as a semver violation.

#### Software version

cargo-semver-checks 0.34.0

#### Operating system

Linux 6.6.47

#### Command-line

```bash
/nix/store/4j0iyaxn6img53nhhpbx165irrl3r0ry-cargo-semver-checks-0.34.0/bin/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.80.1 (376290515 2024-07-16)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

### Build Configuration

```console
$ printenv | rg ^CARGO_
CARGO_HOME=/home/jalil/.local/share/cargo
```

Repository is a `cargo new --lib` blank repo

### Additional Context

This came up while discussing changes to `rustdoc-types` c:

**Comment by obi1kenobi** [comment]
Nice, thank you.

I think the best way to handle this would probably be designing some schema to express `pub type` aliases, and flag it that way. It would let us scan for it together with any other changes to what users of the `pub type` can do, instead of special-casing it for "is this struct used as part of any `pub type` aliases" etc.

I'm not yet sure what the schema for this should look like. Figuring that out is the first action item here.
